src/components/generator/GeneratorPage.js

```diff
@@ -84,39 +85,68 @@ const GeneratorPage = () => {
+                            <StyledSingleTableCell>Edit</StyledSingleTableCell>
+                            <StyledSingleTableCell>Generate</StyledSingleTableCell>
+                        </StyledHeadTableRow>
+                    </StyledTableHead>
+
+                    <TableBody>
                     {availableProjects.map((project) => (
-                        <BodyTableRow key={project.id}>
-                            <CellSingle>{project.name}</CellSingle>
-                            <CellDouble>{project.description}</CellDouble>
-                            <CellSingle>{new Date(project.created_at).toLocaleString("de-DE")}</CellSingle>
-                            <CellSingle>{new Date(project.modified_at).toLocaleString("de-DE")}</CellSingle>
-                            <GenerateButton onClick={() => Notifications.notify("Not implemented yet", "warning")}>
-                                Generate | Edit
-                            </GenerateButton>
-                        </BodyTableRow>
+                        <StyledBodyTableRow key={project.id}>
+                            <StyledSingleTableCell>{project.name}</StyledSingleTableCell>
+                            <StyledDoubleTableCell>{project.description}</StyledDoubleTableCell>
+                            <StyledSingleTableCell>{new Date(project.created_at).toLocaleString("de-DE")}</StyledSingleTableCell>
+                            <StyledSingleTableCell>{new Date(project.modified_at).toLocaleString("de-DE")}</StyledSingleTableCell>
+                            <StyledSingleTableCell>
+                              <StyledLink to={`${project.id}`}>
+                                <StyledButton>
+                                  Edit
+                                </StyledButton>
+                              </StyledLink>
+                            </StyledSingleTableCell>
+                            <StyledSingleTableCell>
+                              <StyledButton onClick={() => onGenerate(project)}>
+                                  Generate
+                              </StyledButton>
+                            </StyledSingleTableCell>
+                        </StyledBodyTableRow>
                     ))}
-                    </tbody>
-                </TableRoot>
+                    </TableBody>
+                </StyledTable>
+              </TableContainer>
             ) : projectRetrievalState === "success" ? (
                 <p>There are no projects yet</p>
             ) : projectRetrievalState === "loading" &&
                 <p>Loading...</p>
             }
+            {(generate === true) ? (
+              <GenerateDialog
+                setGenerate={setGenerate}
+                open={open}
+                setOpen={setOpen}
+                selectedProject={selectedProject}
+                setSelectedProject={setSelectedProject}
+              />
+            ) : <div></div>}
```

Comment:
```suggestion
            {selectedProject && (
              <GenerateDialog
                setGenerate={setGenerate}
                open={open}
                setOpen={setOpen}
                selectedProject={selectedProject}
                setSelectedProject={setSelectedProject}
              />
            }
```
`generate` state is redundant and can be removed. `selectedProject` will always evaluate to the same boolean value as `generate`.

Reply:
implemented in branch: "generatorpage-add-delete"